### DISCUSSION TITLE
Construct Email with raw_body

### DISCUSSION THREAD

**Discussion by cw789** [swoosh/swoosh#623] [General]
I'm thinking about writing a security oriented newsletter system with PGP signed emails (possibly based on Swoosh as I've started to like it, since it will land in Phoenix v1.6).

But I don't see an option to construct the Email with `raw_body`.
Do I miss something?

**Comment by princemaple** [comment]
Is it something generally supported by the service providers? Most of them take json payload, how would that work? Does it only work with SMTP?

**Reply by cw789** [reply]
Thats a good point. I never scratched my head with other service providers. Just happily alwas using SMTP for now.

So ok it then would probably only work with SMTP. In this case I won't bother making this cabability a thing for Swoosh.

What do you think?

**Reply by princemaple** [reply]
This probably needs to be a separate thing built on top of `:gen_smtp`. If built, we can try and integrate it into smtp adapter.

**Reply by cw789** [reply]
Ok. If I will go for it, I would came back then.

**Reply by princemaple** [reply]
👍
